File: client-java/controller/src/main/java/org/evomaster/client/java/controller/dynamodb/operations/TypeOperation.java
package org.evomaster.client.java.controller.dynamodb.operations;

import org.evomaster.client.java.controller.dynamodb.DynamoDbAttributeType;

/**
 * DynamoDB {@code attribute_type(path, type)} predicate operation.
 */
public class TypeOperation extends QueryOperation {

    private final String fieldName;
    private final DynamoDbAttributeType expectedType;

    /**
     * Creates a type operation.
     *
     * @param fieldName field name coming from DynamoDB expression/condition
     * @param expectedType expected DynamoDB attribute type
     */
    public TypeOperation(String fieldName, DynamoDbAttributeType expectedType) {
        this.fieldName = fieldName;
        this.expectedType = expectedType;
    }

    /**
     * @return field name coming from DynamoDB expression/condition
     */
    public String getFieldName() {
        return fieldName;
    }

    /**
     * @return expected DynamoDB attribute type
     */
    public DynamoDbAttributeType getExpectedType() {
        return expectedType;
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/operations/TypeOperation.kt
package org.evomaster.client.java.controller.mongo.operations

import org.bson.BsonType

/**
 * Represent $type operation.
 * Selects documents where the value of the field is an instance of the specified BSON type(s).
 */
open class TypeOperation (open val fieldName: String, open val type: BsonType) : QueryOperation()